src/app/src/main/kotlin/org/imperial/mrc/hint/logic/UserLogic.kt

```diff
@@ -74,10 +74,10 @@ class DbProfileServiceUserLogic(private val userRepository: UserRepository,
             throw UserException("invalidEmail")
         }
 
-        val caseInsensitiveDomainRegex = Regex("(?-i)${emailArray[0]}@(?i)${emailArray[1]}")
+        val caseInsensitiveEmail = Regex("(?i)${email}")
```

Comment:
works well, but this regex doesn't check If this is indeed a valid email address format ? Is this being checked somewhere else?

Reply:
The lines immediately above do some basic validation, throwing an error if there aren't two strings around an @, but validation isn't really the role of this function, which is just to check if an email exists in the database. I think there are also some cases where we wouldn't want validation here, such as with guest users. The validation is primarily done when a user is created, I believe.